internal/plugintest/config.go

```diff
@@ -60,11 +66,31 @@ func DiscoverConfig(ctx context.Context, sourceDir string) (*Config, error) {
 
 		logging.HelperResourceTrace(ctx, fmt.Sprintf("Adding potential Terraform CLI source of releases.hashicorp.com exact version %q for installation in: %s", tfVersion, tfDir))
 
-		sources = append(sources, &releases.ExactVersion{
+		if tempDir == "" {
+			tempDir = os.TempDir()
+		}
+		pathParts := []string{
+			strings.TrimRight(tempDir, string(os.PathSeparator)),
+			"plugintest-terraform",
+			strconv.Itoa(os.Getpid()),
+			tfVersion.String(),
+		}
+		tfDir := strings.Join(pathParts, string(os.PathSeparator))
+		if err := os.MkdirAll(tfDir, 0o700); err != nil {
+			return nil, fmt.Errorf("failed to create temporary directory for Terraform CLI: %w", err)
+		}
+		findSource := &fs.ExactVersion{
+			ExtraPaths: []string{tfDir},
+			Product:    product.Terraform,
+			Version:    tfVersion,
+		}
+		releasesSource := &releases.ExactVersion{
 			InstallDir: tfDir,
 			Product:    product.Terraform,
 			Version:    tfVersion,
-		})
+		}
+		sources = append(sources, findSource, releasesSource)
```

Comment:
This is the part that makes this faster. Instead of a "releases source," use a "find source" and a "releases source."